design/pipeline-artifacts.md

```diff
@@ -35,30 +45,140 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 
 ## Downloading artifacts: `download`
 
-`download` is a shortcut for the [Download Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact) task.
+`download` can be _thought of_ as a shortcut for the [Download Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact) task.
+(In reality, it also needs to download older FCS and UNC artifacts seamlessly.)
+It will download artifacts uploaded from a previous job, stage, or from another pipeline.
+
+### Partial schema
+
+Listing just the unique properties of the `download` step:
+```yaml
+- download: string - the pipeline to download from (`current` or a resource id) or the word `none`
+  artifact: string - the specific artifact; optional unless `path` specified
+  patterns: string - minimatch patterns of files to grab from the artifact(s); defaults to **/*
+  path: string - the destination path for where to download the artifact(s)
+```
+
+### Automated download
 
-It will download artifacts uploaded from a previous job or stage or from another pipeline.
+The "magic" of download is how often you _don't_ have to specify it.
+By default, all artifacts produced in the current pipeline and all artifacts in all referenced pipelines will be downloaded.
+As soon as you include an explicit `download` step (or task), the user is in total control - no more automatic download for that job.
+Similarly, you can leave out `artifact`, `patterns`, or `path` to get some automatic behaviors (described below).
+
+Or to avoid downloading any of the artifacts at all:
+
+```yaml
+- download: none
+```
+
+### Choosing which pipelines and artifacts to download
+
+One `download` step will download from one pipeline, either the `current` pipeline or a pipeline's resource ID.
+
+If `artifact` is absent, the step will download all artifacts that are part of the pipeline.
+If `artifact` is present, it's a single artifact name from the pipeline.
 
 ### Artifact download location
 
-Artifacts are downloaded either to `$PIPELINE_RESOURCESDIRECTORY` or to the directory specified in `root`. Each artifact is given its own directory e.g. `$PIPELINE_RESOURCESDIRECTORY\default` for the `default` artifact of this pipeline. Artifacts coming from other Pipelines are each given one directory per pipeline e.g. `$PIPELINE_RESOURCESDIRECTORY\some-other-pipeline\default` for the `default` artifact of the `some-other-pipeline` pipeline.
+#### YAML / unified pipelines
 
-### Schema
+Artifacts are automatically downloaded to the pipeline's workspace by default. We introduce a new variable for unified pipelines: `$(Pipeline.Workspace)`. `Pipeline.Workspace` is one level up from `$(System.DefaultWorkingDirectory)`. This is the "shallowest" place where artifacts may be placed.
+
+The spec that follows is lengthy because it covers details and edge cases.
```

Comment:
@ericsciple in lieu of a table, I boiled this down to its 3-bullet-point essence. I think you'll approve.